### ISSUE TITLE
Persistent connections are not faster then regular connections ?

### ISSUE THREAD

**Issue by grosser** [lostisland/faraday#776]
```
require 'faraday'

def bench(name)
  x = Benchmark.realtime do
    100.times do
      yield
    end
  end
  puts "#{name}: #{x}"
end

bench("normal") do
  Faraday.get("https://api.github.com/rate_limit", {}, authorization: "token #{ENV["GITHUB_TOKEN"]}")
end

conn = Faraday.new("https://api.github.com") do |r|
  r.adapter :net_http_persistent
end
bench("persistent") do
  conn.get("/rate_limit", {}, authorization: "token #{ENV["GITHUB_TOKEN"]}")
end
http = Net::HTTP::Persistent.new
bench("raw") do
  http.request URI "https://api.github.com/rate_limit"
end

```

```
normal: 28.042679999955
persistent: 28.48966299998574
raw: 17.34108699997887
```

am I doing something wrong or is that expected ?

**Comment by iMacTia** [comment]
Hi @grosser,

I believe someone raised at some point the issue that the current adapter for `Net::HTTP::Persistent` is re-creating the connection on every request, which of course doesn't make any sense...

I promised myself I'd look into this while moving it outside into a separate gem for v1.0, but if you want to try a quick-fix then it's more than welcome

**Comment by iMacTia** [comment]
This was addressed in #778
